src/osm_parser.rs

```diff
@@ -37,8 +37,8 @@ fn lat_lon_to_minecraft_coords(
     let (min_lon, min_lat, max_lon, max_lat) = bbox;
 
     // Calculate the relative position within the bounding box
-    let rel_x: f64 = (lon - min_lon) / (max_lon - min_lon);
-    let rel_z: f64 = (lat - min_lat) / (max_lat - min_lat);
+    let rel_x: f64 = 1.0 - (lat - min_lat) / (max_lat - min_lat);
+    let rel_z: f64 = (lon - min_lon) / (max_lon - min_lon);
```

Comment:
I'd like to swap around the (z, x) a few lines below, and adjust this code accordingly, but it'll depend on what exactly scale_factor_x and scale_factor_z do. With the current version of the code, it seems like we're using scale_factor_z to scale the x coordinate and vice versa